src/languages/fsharp.js

```diff
@@ -221,16 +225,130 @@ export default function(hljs) {
     ]
   };
 
-  // 'a or ^a
+  // Most identifiers can contain apostrophes
+  const IDENTIFIER_RE = /[a-zA-Z_](\w|')*/;
+
+  const QUOTED_IDENTIFIER = {
+    scope: 'variable',
+    begin: /``/,
+    end: /``/
+  };
+
+  // 'a or ^a where a can be a ``quoted identifier``
+  const BEGIN_GENERIC_TYPE_SYMBOL_RE = /\B('|\^)/
   const GENERIC_TYPE_SYMBOL = {
-    match: regex.concat(/('|\^)/, hljs.UNDERSCORE_IDENT_RE),
     scope: 'symbol',
+    variants: [
+      // the type name is a quoted identifier:
+      { match: regex.concat(BEGIN_GENERIC_TYPE_SYMBOL_RE, /``.*?``/) },
+      // the type name is a normal identifier (we don't use IDENTIFIER_RE because there cannot be another apostrophe here):
+      { match: regex.concat(BEGIN_GENERIC_TYPE_SYMBOL_RE, hljs.UNDERSCORE_IDENT_RE) }
+    ],
+    relevance: 0
+  };
+
+  // List or symbolic operator characters from the FSharp Spec 4.1, minus the dot, and with `?` added, used for nullable operators.
+  const OPERATOR_CHARS = Array.from("!%&*+-/<=>@^|~?");
+  const OPERATOR_CHAR_RE = regex.concat('[', ...OPERATOR_CHARS.map(regex.escape), ']');
+  // TODO: choose one of the alternative definitions:
+  // const OPERATOR_CHAR_RE = regex.either(...OPERATOR_CHARS.map(regex.escape));
+  // const OPERATOR_CHAR_RE = /[!%&*+\-\/<=>@^|~?]/;
+  // The lone dot operator is special. It cannot be redefined, and we don't want to highlight it. It can be used as part of a multi-chars operator though.
+  const OPERATOR_CHAR_OR_DOT_RE = regex.either(OPERATOR_CHAR_RE, /\./);
+  // When a dot is present, it must be followed by another operator char:
+  const OPERATOR_FIRST_CHAR_OF_MULTIPLE_RE = regex.concat(OPERATOR_CHAR_OR_DOT_RE, regex.lookahead(OPERATOR_CHAR_OR_DOT_RE));
```

Comment:
So is `..` intended to be valid here?

Reply:
The second part of `OPERATOR_FIRST_CHAR_OF_MULTIPLE_RE ` is a look-ahead, but the goal is to match `.` when followed by any operator char, so `..` must be matched yes. (It's the built-in operator to define ranges, e.g. `[1..10]`)